File: app/src/main/java/rod/bailey/trafficatnsw/util/EmptyListEmptyMessagePredicate.java
package rod.bailey.trafficatnsw.util;

import android.widget.ListAdapter;

public class EmptyListEmptyMessagePredicate implements IEmptyMessagePredicate {

	@Override
	public boolean showEmptyMessage(ListAdapter adapter) {
		return (adapter.getCount() == 0);
	}

}

File: app/src/main/java/rod/bailey/trafficatnsw/ui/predicate/EmptyListEmptyMessagePredicate.kt
package rod.bailey.trafficatnsw.ui.predicate

import android.widget.ListAdapter

/**
 * Basic definition of IEmptyMessagePredicate that returns true iff the
 * list is devoid of items.
 */
class EmptyListEmptyMessagePredicate : IEmptyMessagePredicate {
	override fun showEmptyMessage(adapter: ListAdapter): Boolean {
		return adapter.count == 0
	}
}
